modules/features2d/src/fast.cpp

```diff
@@ -425,6 +485,9 @@ void FAST(InputArray _img, std::vector<KeyPoint>& keypoints, int threshold, bool
     CV_OCL_RUN(_img.isUMat() && type == FastFeatureDetector::TYPE_9_16,
                ocl_FAST(_img, keypoints, threshold, nonmax_suppression, 10000));
 
+    cv::Mat img = _img.getMat();
+    CALL_HAL(fast, hal::FAST, img, keypoints, threshold, nonmax_suppression, type);
```

Comment:
Is it possible to integrate [carotene implementation](https://github.com/opencv/opencv/blob/dcdd6af5a856826fe62c95322145731e702e54c5/3rdparty/carotene/include/carotene/functions.hpp#L1913-L1916) using this interface?

Reply:
No, we can only do something like this:

* add one more interface for FAST

```cpp
inline int hal_ni_FAST(const uchar* src_data, size_t src_step, int width, int height,
                       uchar* keypoints_data, size_t* keypoints_size,
                       int threshold, bool nonmax_suppression, int type) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }

//! @cond IGNORED
#define cv_hal_FAST hal_ni_FAST
//! @endcond
```
* try to call this interface one by one
```cpp
    cv::Mat img = _img.getMat();
    CALL_HAL(fast_dense, hal::FAST, img, keypoints, threshold, nonmax_suppression, type);

    size_t keypoints_size;
    CALL_HAL(fast, cv_hal_FAST, img.data, img.step, img.cols, img.rows,
             (uchar*)(keypoints[0]), &keypoints_size, threshold, nonmax_suppression, type);
```

@vpisarev, if it isn't acceptable, could we discuss it one more time?